pkg/controller/common/license/parser.go

```diff
@@ -12,9 +12,30 @@ import (
 
 func ParseEnterpriseLicense(raw map[string][]byte) (EnterpriseLicense, error) {
 	var license EnterpriseLicense
-	err := json.Unmarshal(raw[FileName], &license)
+
+	bytes, err := FetchLicenseData(raw)
+	if err != nil {
+		return license, err
+	}
+
+	err = json.Unmarshal(bytes, &license)
 	if err != nil {
-		return EnterpriseLicense{}, errors.Wrapf(err, "license cannot be unmarshalled")
+		return license, errors.Wrapf(err, "license cannot be unmarshalled")
 	}
+
 	return license, nil
 }
+
+func FetchLicenseData(raw map[string][]byte) ([]byte, error) {
+	if len(raw) != 1 {
+		return nil, errors.New("Elasticsearch license secret needs to contain exactly one file with any name")
```

Comment:
This is can be an Enterprise license secret as well as an Elasticsearch license secret, needs rewording to 'license secret' maybe.

Reply:
Ah, good catch, fixed.